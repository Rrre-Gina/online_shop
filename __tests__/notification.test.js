import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Notification from '../components/notification';

const close = () => null;

it("render notification component", () => {
    render(<Notification text='Ошибка 500, попробуйте еще раз' close={ close } />);
    const notification = screen.getByTestId('notification');
    expect(notification).toHaveClass('notification');
    expect(notification).toHaveTextContent('Ошибка 500, попробуйте еще раз');

    const tree = renderer.create(<Notification text='Ошибка 500, попробуйте еще раз' close={ close } />).toJSON();
    expect(tree).toMatchSnapshot();
});
