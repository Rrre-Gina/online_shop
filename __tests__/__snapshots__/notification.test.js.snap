// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render notification component 1`] = `
<div
  className="notification"
  data-testid="notification"
>
  <button
    className="modal-close"
    onClick={[Function]}
  >
    ✖
  </button>
  Ошибка 500, попробуйте еще раз
</div>
`;
